src/metrics_server/model.ts

```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 The Outline Authors
```

Comment:
Following Domain-Driven Design, I like to reserve the [(domain) model](https://www.martinfowler.com/eaaCatalog/domainModel.html) to be product concepts, and product concepts only. Things that we communicate to the user.

In this case, the reports would make sense to be here: HourlyConnectionMetricsReport and deps, DailyFeatureMetricsReport and deps.

Notice that the Row types wouldn't belong here: they are an implementation (serialization) detail the user doesn't care about.

The InsertableTable is also not a product concept. It's more like a infrastructure concepts that is loosely tied to the application. I found it helpful to place those potentially app-independent concepts in a separate `infrastructure/` folder. So we could have this under `metrics/server/infrastructure/table.ts`

Reply:
Thanks for explaining. I moved the file to `metrics/server/infrastructure/table.ts`. 

Would it be okay to leave the `HourlyConnectionMetricsReport` and `DailyFeatureMetricsReport` in their respective files, or would it be preferrable to move them to `model.ts`?

Reply:
I think it would be better to put them in a model.ts, which then becomes product documentation. We can refer to it to see all we log. But it's up to you.

Reply:
Done in 05c80c2.